avocado/core/loader.py

```diff
@@ -694,8 +693,7 @@ def ignore_broken(klass, uid, params=None):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest,
-                                           pipes.quote(test_path))
+                    return self._make_test(test.SimpleTest, test_path)
```

Comment:
The current state of master is a similar to the old "remote copy" feature, that is, it has caveats, and even worse, undocumented ones.

So, while the current state of master allows for one use case, it is inconsistent with others. I suggest we take the path of simplicity, at the cost of one extra feature.

I suggest we drop altogether the support for passing parameters, so that every test reference, in the case of being resolved by simple tests, maps to a path (alone).  Example:

* `foo bar` is an executable
* `foo` is an executable
* `avocado run "foo bar"` *always run `foo bar`, never runs `foo` with argument `bar`

Unless there's a major use case supporting passing arguments to executables as a test reference, I suggest we go this route.

A more confusing and costlier to implement solution (that I don't recomend, but that would be still consistent), would be to support partial matches for the path, with a given set of precedence.  Example:

* `foo` is an executable
* `foo bar` *does not* exist
* `avocado run "foo bar"` breaks the test reference and considers: executable = `foo`, arguments = `bar`.
* `foo bar` is created as an executable
* `avocado run "foo bar"` finds exact match first, and considers: executable = `foo bar`, arguments = `` (empty).

This results in wasted brain cycles for our users, and maybe even a bunch of "WTFs" from them.  Creating a script with test parameters built in is so much easier compared to supporting this.  Also, we already support passing parameters (including mux ones) as environment variables which executables can use.